src/app/routes/connector-ui/contract-agreement-page/contract-agreement-cards/contract-agreement-cards.component.html

```diff
@@ -58,18 +59,35 @@
           </div>
         </div>
       </div>
-      <!-- Counterparty Address -->
-      <div class="flex flex-col flex-grow">
-        <div class="text-slate text-sm uppercase tracking-wider">
-          Other Connector
+      <div class="flex flex-row space-x-[10px] justify-between">
+        <!-- Counterparty Address -->
+        <div class="flex flex-col">
+          <div class="text-slate text-sm uppercase">Other Connector</div>
+          <div>
+            <a
+              class="link break-all"
+              target="_blank"
+              [attr.href]="card.counterPartyAddress"
+              >{{ card.counterPartyAddress }}</a
+            >
+          </div>
         </div>
-        <div>
-          <a
-            class="link"
-            target="_blank"
-            [attr.href]="card.counterPartyAddress"
-            >{{ card.counterPartyAddress }}</a
-          >
+        <!-- Status -->
+        <div
+          *ngIf="card.isConsumingLimitsEnforced"
+          class="flex flex-col"
+          [matTooltip]="card.statusTooltipText">
+          <div class="text-slate text-sm uppercase tracking-wider text-right">
+            Status
+          </div>
+          <div class="text-right">
+            {{ card.statusText }}
+            <mat-icon
+              *ngIf="card.statusText === 'Inactive'"
```

Comment:
this is too flaky. we need a flag. the statusText was only meant for the label. so if we need to use this "truth" to decide other things, we still need the original flag of status: 'INACTIVE' | 'ACTIVE' | 'PROVIDING'